doc/python-guidelines.md

```diff
@@ -0,0 +1,7 @@
+# Python Guidelines
+
+Few of them:
+
+* Use type annotations from `typing` module and `mypy` for static verification
+* Deliver PEP8 formatted code. Use linters to check and auto-format tools to fix.
+* In case of uncertainty regarding style and approach use Google Python style-guide: https://google.github.io/styleguide/pyguide.html
```

Comment:
Everyone should add a comment in a brief of what exactly that module or function does. This helps to iterate over the code at later point of time easily irrespective of who has actually written that code.